### ISSUE TITLE
I try the int8_support branch but get an error.

### ISSUE THREAD

**Issue by aininot260** [NVIDIA-AI-IOT/torch2trt#169]
AttributeError: type object 'tensorrt.tensorrt.CalibrationAlgoType' has no attribute 'ENTROPY_CALIBRATION_2'
My TensorRT version on NVIDIA Jetson Xavier is 5.0.6+cuda10.0.
My Jetpack version is 4.2.
It doesn't have the attribute 'ENTROPY_CALIBRATION_2'.
How can I use the int8_mode in this situation?
Thanks.

**Comment by jaybdub** [comment]
Hi aininot260,

Thanks for reaching out!

I would guess this calibration mode isn't supported in TensorRT 5.0.6.

Could you try setting ``int8_calib_algorithm=trt.CalibrationAlgoType.ENTROPY_CALIBRATION``?

By the way, the branch is *highly* subject to change until it is merged into master.  That said, the feedback is helpful :)

Please let me know if you have any questions or if that works for you.

Best,
John

**Comment by aininot260** [comment]
I try:
    self.model = torch2trt(self.model,[images],int8_mode=True,int8_calib_algorithm=trt.CalibrationAlgoType.ENTROPY_CALIBRATION)

but also get:

Traceback (most recent call last):
  File "main.py", line 8, in <module>
    from detectors.detector_factory import detector_factory
  File "./src/lib/detectors/detector_factory.py", line 6, in <module>
    from .ctdet import CtdetDetector
  File "./src/lib/detectors/ctdet.py", line 8, in <module>
    from torch2trt import torch2trt
  File "/usr/local/lib/python3.6/dist-packages/torch2trt/__init__.py", line 1, in <module>
    from .torch2trt import *
  File "/usr/local/lib/python3.6/dist-packages/torch2trt/torch2trt.py", line 5, in <module>
    from .calibration import TensorBatchDataset, DatasetCalibrator
  File "/usr/local/lib/python3.6/dist-packages/torch2trt/calibration.py", line 17, in <module>
    class DatasetCalibrator(trt.IInt8Calibrator):
  File "/usr/local/lib/python3.6/dist-packages/torch2trt/calibration.py", line 19, in DatasetCalibrator
    def __init__(self, inputs, dataset, batch_size=1, algorithm=trt.CalibrationAlgoType.ENTROPY_CALIBRATION_2):
AttributeError: type object 'tensorrt.tensorrt.CalibrationAlgoType' has no attribute 'ENTROPY_CALIBRATION_2'

May be the 'ENTROPY_CALIBRATION_2' in calibration.py need to change?

I modify the calibration.py in source code and reinstall,but also get the error:

Traceback (most recent call last):
  File "main.py", line 15, in <module>
    detector = detector_factory[opt.task](opt)
  File "./src/lib/detectors/ctdet.py", line 28, in __init__
    super(CtdetDetector, self).__init__(opt)
  File "./src/lib/detectors/base_detector.py", line 32, in __init__
    self.model = torch2trt(self.model,[images],int8_mode=True,int8_calib_algorithm=trt.CalibrationAlgoType.ENTROPY_CALIBRATION)
  File "/usr/local/lib/python3.6/dist-packages/torch2trt/torch2trt.py", line 399, in torch2trt
    engine = builder.build_cuda_engine(network)
TypeError: read_calibration_cache() takes 1 positional argument but 2 were given

**Comment by jaybdub** [comment]
Hmm,

The error indicates that the cuda engine builder is attempting to call [read_calibration_cache](https://github.com/NVIDIA-AI-IOT/torch2trt/pull/168/files#diff-ad03ff1ec0887e8e18bbbe17ecec12baR59) with more arguments.

There is likely an API discrepancy between TensorRT 5.0 and 5.1.  

You could try modifying the method definition to take variable arguments.

```python
    def read_calibration_cache(self, *args, **kwargs):
    
    def write_calibration_cache(self, cache, *args, **kwargs):
```

In reality, I need to test against ``5.0`` before I can guarantee.  Are you able to use TensorRT 5.1+?

Best,
John

**Comment by aininot260** [comment]
Thank you very much!My problem is solved.
It is challenging to change TensorRT 5.0 to TensorRT5.1+.I can only find aarch64 version of TensorRT in Jetpack. Updating the jetpack version means reinstalling the operating system.

**Comment by jaybdub** [comment]
I see.  Thanks for sharing.  Just to clarify,

Making this change:

```python
    def read_calibration_cache(self, *args, **kwargs):
    
    def write_calibration_cache(self, cache, *args, **kwargs):
```

Allowed you to convert and calibrate your model with TensorRT 5.0?

Best,
John

**Comment by aininot260** [comment]
Yes.
I also modify the 'ENTROPY_CALIBRATION_2' in calibration.py,change it into 'ENTROPY_CALIBRATION'.

**Comment by jaybdub** [comment]
Okay great, thanks.

I'll consider adding these changes.

Best,
John

**Comment by jaybdub** [comment]
FYI, I believe I've fixed support for TensorRT 5.0 in int8_branch.

https://github.com/NVIDIA-AI-IOT/torch2trt/pull/168/commits/8e6e475db78061a815665ed7a00afbcc289def30

Please let me know if it works for you.

Best,
John

**Comment by aininot260** [comment]
It works well.
Thanks!
